Add language identification recipe using the Voxlingua107 dataset

alumae opened the pull request:
This PR adds a recipe for training a spoken language identification model using the VoxLingua107 dataset (http://bark.phon.ioc.ee/voxlingua107/). It is fairly similar to CommonLanguage and VoxCeleb recipes, except that it uses a WebDataset based data loading pipeline. A model trained using this recipe is available at https://huggingface.co/TalTechNLP/voxlingua107-epaca-tdnn.

mravanelli commented:
Hi @aelimame,
thank you for this PR. What is the accuracy you get?

alumae commented:
This receipe results in accurcy of around 7.3% of VoxLingua107 dev set. However, I have a newer slightly changed recipe that gives a better accuracy (6.7%)and also better quality embeddings for finetuning. I'll change the PR accordingly, so please don't merge it yet.

mravanelli commented:
Hi @aelimame, any update on that?

aelimame commented:
> Hi @aelimame, any update on that?

Hi @mravanelli I think you got the user name wrong, I guess you wanted to notify @alumae :)

mravanelli commented:
oops....sorry

On Fri, 10 Sept 2021 at 20:56, aelimame ***@***.***> wrote:

> Hi @aelimame <https://github.com/aelimame>, any update on that?
>
> Hi @mravanelli <https://github.com/mravanelli> I think you got the user
> name wrong, I guess you wanted to notify @alumae
> <https://github.com/alumae> :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/speechbrain/speechbrain/pull/950#issuecomment-917308195>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEA2ZVSQFKBRMMGV3CZETNLUBKSKNANCNFSM5CW5USUA>
> .
>

alumae commented:
I made some small changes to the recipe and documentation. Error rate on VoxLingua107 dev set is now 6.7%. It should be ready for merge.

mravanelli commented:
Super cool! ok, I think we can merge it right after we replicated the
experiment. Thank you very much!

On Mon, 13 Sept 2021 at 09:46, Tanel Alumäe ***@***.***>
wrote:

> I made some small changes to the recipe and documentation. Error rate on
> VoxLingua107 dev set is now 6.7%. It should be ready for merge.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/speechbrain/speechbrain/pull/950#issuecomment-918207914>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEA2ZVXK7MTE3MIPL22MGPTUBX6EXANCNFSM5CW5USUA>
> .
>

mravanelli commented:
Thank you  @alumae. This is a great contribution! I tried the recipe with the following command:

`python train.py hparams/train_epaca_tdnn_ce_wds.yaml --shards_url=http://bark.phon.ioc.ee/lw/korpused/voxlingua107/shards/ --shard_cache_dir=/miniscratch/ravanelm/voxlingua107`

It looks like there is an issue with webdataset (I installed the latest version):

```
  File "/network/tmp1/ravanelm/speechbrain_github/speechbrain_langid_huge/recipes/VoxLingua107/lang_id/train.py", line 193, in dataio_prep_shards
    wds.WebDataset(
TypeError: WebDataset() got an unexpected keyword argument 'length'
```

Do you see the same with the latest version of webdataset?  @Gastron, any idea?

Minor: 
- train_epaca_tdnn_ce_wds.yaml => train_ecapa.yaml
- Add year close to your name in the yaml file
- add file called "extra-dependencies.txt" with the webdataset dependency

alumae commented:
I made the code compatible to the latest webdataset version and also submitted the small changes suggested by Mirco.

mravanelli commented:
Thank you! Let me try it again.

On Tue, 14 Sept 2021 at 10:16, Tanel Alumäe ***@***.***>
wrote:

> I made the code compatible to the latest webdataset version and also
> submitted the small changes suggested by Mirco.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/speechbrain/speechbrain/pull/950#issuecomment-919195224>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEA2ZVVFQNUIO6V6X3NLSRTUB5KMXANCNFSM5CW5USUA>
> .
>

mravanelli commented:
Hi,
I'm trying to train the model with the shards downloaded from the web. I'm trying with python 3.9 and I have the following error:
```
tarfile.ReadError: ('truncated header', <_io.BufferedReader name='/miniscratch/ravanelm/voxlingua107/31dd67bb-a432-3593-8516-69f521f5c314'>, {})
tarfile.TruncatedHeaderError: truncated header
```
I have the same error with python 3.8. However, it I set `num_workers: 1` it seems to work.  Do you see the same issue?

It looks that in this case, training a single epoch will take 14 hours rather than 40 minutes (as mentioned in the readme). Do you expect that when downloading the shards from the web?

I'm now trying to download the shards manually and follow option2.

Thank you  for your help!

alumae commented:
When training directly on the data downloaded from the web, then the 1st epoch will take more for sure. The next epochs will be faster, as the tar files are cached.

However, I now also run into all sort of problems when training directly on the web data, even with `num_workers: 1`:

```
  File "/ceph/home/tanel.alumae/miniconda3/envs/speechbrain/lib/python3.8/site-packages/webdataset/gopen.py", line 81, in read                                                                                                                                                            
    result = self.stream.read(*args, **kw)            
ValueError: ('read of closed file', <webdataset.shardcache.CacheStream object at 0x7fba9ea7e6d0>, {}) 
```

I suspect there is a problem with a newer version of webdataset, as it used to work for me with an older version.

I'll investigate this, but we might have to remove option 3.

mravanelli commented:
Sure, it makes sense. Let's see if we can make option 3 working.

I'm now trying option 2 now. I downloaded all the shards available in `http://bark.phon.ioc.ee/lw/korpused/voxlingua107/shards/train/`. However, it looks like some shards are missing. For instance, when I start training, it gives an error because it cannot find` /train/shard-000195.tar` (which is not present in the sever as well). It looks like there are many other shards missing. Is there something to fix somewhere?

minor: `hparams/train_epaca.yaml` => `hparams/train_ecapa.yaml`

Gastron commented:
I suppose the missing shards might lead to the errors on option 3 as well?

Though I would not worry much about getting option 3 to work. That kind of on-the-fly downloading is maybe useful for debugging, but in a ready-made recipe it's probably a waste of GPU-time, right?

alumae commented:
The missing shards was my bad, I don't know how it happened, sorry about that. It's fixed now.
You should be able to download the missing shards using the `--no-clobber` option of wget, smth like:
```
wget  --no-clobber -r -nH --cut-dirs=4 --no-parent --reject="index.html*" http://bark.phon.ioc.ee/lw/korpused/voxlingua107/shards/
```

mravanelli commented:
Thank you @alumae. It seems working now. I'm training a model to see if I can replicate your numbers. According to the documentation, I see that an epoch should take 40 minutes on a NVIDIA A100. On my side, it looks like it takes 4 hours on a RTX8000. The difference sounds too much to me. What do you think?

mravanelli commented:
I spoke too early. It looks like there is an issue in one of the shards:

`tarfile.ReadError: ("unexpected end of data @ <_io.BufferedReader name='/miniscratch/ravanelm/voxlingua107/voxlingua107_shards//train/shard-000165.tar'>", <_io.BufferedReader name='/miniscratch/ravanelm/voxlingua107/voxlingua107_shards//train/shard-000165.tar'>, {'__url__': '/miniscratch/ravanelm/voxlingua107/voxlingua107_shards//train/shard-000165.tar', '__worker__': '(1, 4)', '__rank__': 'None', '__nodeinfo__': "('cn-c017', 94144)"})`

alumae commented:
There was indeed a problem with shard-000165.tar. Everything should be fixed now. Sorry about that.

Regarding training time: are the shards on a local disk or network share? If they are on a network drive, it could become a bottleneck, even when using this tar-file based reading which should be much more network-friendly than reading individual wav files.

mravanelli commented:
Thank you, let me try it now. Yes, it is on the shared filesystem because
the dataset is bigger than the local SSD. It could indeed be a bottleneck,
especially on LUSTRE filesystems. However, I did some experiments in the
past on the same filesystem using webdataset and that wasn't a big
bottleneck. I guess Aku has some experience to share as well with that.

On Mon, 20 Sept 2021 at 11:33, Tanel Alumäe ***@***.***>
wrote:

> There was indeed a problem with shard-000165.tar. Everything should be
> fixed now. Sorry about that.
>
> Regarding training time: are the shards on a local disk or network share?
> If they are on a network drive, it could become a bottleneck, even when
> using this tar-file based reading which should be much more
> network-friendly than reading individual wav files.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/speechbrain/speechbrain/pull/950#issuecomment-923038081>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEA2ZVS6FKOKPMBLCVYVP2TUC5H4TANCNFSM5CW5USUA>
> .
>

alumae commented:
I just rechecked that when the shards are on a local SSD, one iteration really takes around 1:40 with single A100. It shows a much longer ETA at the very beginning (for a few hundred batches), as I guess webdataset has to load many shards to do proper online shuffling.

mravanelli commented:
It looks like it is running now. I will leave it running for the next few days to see if I can replicate the best numbers. I will keep you updated!

mravanelli commented:
Hi @alumae,
I finished the training and I got:

```
epoch: 1, lr: 1.00e-03 - train loss: 2.65 - valid loss: 7.66e-01, valid ErrorRate: 2.24e-01
epoch: 2, lr: 9.69e-04 - train loss: 1.71 - valid loss: 5.08e-01, valid ErrorRate: 1.57e-01
epoch: 3, lr: 9.38e-04 - train loss: 1.46 - valid loss: 4.79e-01, valid ErrorRate: 1.39e-01
epoch: 4, lr: 9.07e-04 - train loss: 1.32 - valid loss: 4.24e-01, valid ErrorRate: 1.28e-01
epoch: 5, lr: 8.76e-04 - train loss: 1.22 - valid loss: 3.74e-01, valid ErrorRate: 1.16e-01
epoch: 6, lr: 8.45e-04 - train loss: 1.16 - valid loss: 3.85e-01, valid ErrorRate: 1.20e-01
epoch: 7, lr: 8.14e-04 - train loss: 1.10 - valid loss: 3.63e-01, valid ErrorRate: 1.04e-01
epoch: 8, lr: 7.83e-04 - train loss: 1.06 - valid loss: 3.43e-01, valid ErrorRate: 1.01e-01
epoch: 9, lr: 7.52e-04 - train loss: 1.02 - valid loss: 3.14e-01, valid ErrorRate: 9.42e-02
epoch: 10, lr: 7.21e-04 - train loss: 9.86e-01 - valid loss: 3.56e-01, valid ErrorRate: 1.05e-01
epoch: 11, lr: 6.90e-04 - train loss: 9.34e-01 - valid loss: 3.80e-01, valid ErrorRate: 1.09e-01
epoch: 12, lr: 6.59e-04 - train loss: 9.17e-01 - valid loss: 3.08e-01, valid ErrorRate: 8.67e-02
epoch: 13, lr: 6.28e-04 - train loss: 8.97e-01 - valid loss: 3.13e-01, valid ErrorRate: 9.23e-02
epoch: 14, lr: 5.97e-04 - train loss: 8.26e-01 - valid loss: 3.18e-01, valid ErrorRate: 8.48e-02
epoch: 15, lr: 5.66e-04 - train loss: 8.46e-01 - valid loss: 3.06e-01, valid ErrorRate: 9.11e-02
epoch: 16, lr: 5.34e-04 - train loss: 8.22e-01 - valid loss: 2.86e-01, valid ErrorRate: 8.16e-02
epoch: 17, lr: 5.03e-04 - train loss: 8.05e-01 - valid loss: 3.16e-01, valid ErrorRate: 9.68e-02
epoch: 18, lr: 4.72e-04 - train loss: 7.81e-01 - valid loss: 3.08e-01, valid ErrorRate: 9.80e-02
epoch: 19, lr: 4.41e-04 - train loss: 7.63e-01 - valid loss: 2.68e-01, valid ErrorRate: 7.53e-02
epoch: 20, lr: 4.10e-04 - train loss: 7.39e-01 - valid loss: 2.94e-01, valid ErrorRate: 8.60e-02
epoch: 21, lr: 3.79e-04 - train loss: 7.17e-01 - valid loss: 3.24e-01, valid ErrorRate: 9.04e-02
epoch: 22, lr: 3.48e-04 - train loss: 7.00e-01 - valid loss: 2.78e-01, valid ErrorRate: 8.03e-02
epoch: 23, lr: 3.17e-04 - train loss: 6.53e-01 - valid loss: 2.94e-01, valid ErrorRate: 7.72e-02
epoch: 24, lr: 2.86e-04 - train loss: 6.56e-01 - valid loss: 2.92e-01, valid ErrorRate: 8.29e-02
epoch: 25, lr: 2.55e-04 - train loss: 6.37e-01 - valid loss: 2.92e-01, valid ErrorRate: 7.97e-02
epoch: 26, lr: 2.24e-04 - train loss: 6.09e-01 - valid loss: 3.01e-01, valid ErrorRate: 9.17e-02
epoch: 27, lr: 1.93e-04 - train loss: 5.93e-01 - valid loss: 2.85e-01, valid ErrorRate: 7.72e-02
epoch: 28, lr: 1.62e-04 - train loss: 5.73e-01 - valid loss: 2.71e-01, valid ErrorRate: 7.27e-02
epoch: 29, lr: 1.31e-04 - train loss: 5.50e-01 - valid loss: 2.73e-01, valid ErrorRate: 7.53e-02
epoch: 30, lr: 1.00e-04 - train loss: 5.30e-01 - valid loss: 2.82e-01, valid ErrorRate: 8.10e-02
```
It looks my numbers are slightly worse than yours, but probably within the normal range of variability. What do you think?
Now, everything looks ready to be merged. I would like to just ask you if you are fine to add the output folder in the official speechbrain gdrive repo and in the official HuggingFace one.

alumae commented:
I actually trained for more epochs (40 instead of 30), forgot to update the yaml file in the repo (I did it now).
Here is my training log:

epoch: 1, lr: 1.00e-04 - train loss: 7.73 - valid loss: 3.31, valid ErrorRate: 2.28e-01
epoch: 2, lr: 9.77e-05 - train loss: 5.59 - valid loss: 2.68, valid ErrorRate: 2.02e-01
epoch: 3, lr: 9.54e-05 - train loss: 4.56 - valid loss: 2.13, valid ErrorRate: 1.87e-01
epoch: 4, lr: 9.31e-05 - train loss: 3.87 - valid loss: 1.66, valid ErrorRate: 1.62e-01
epoch: 5, lr: 9.08e-05 - train loss: 3.40 - valid loss: 1.41, valid ErrorRate: 1.50e-01
epoch: 6, lr: 8.85e-05 - train loss: 3.10 - valid loss: 1.18, valid ErrorRate: 1.32e-01
epoch: 7, lr: 8.62e-05 - train loss: 2.91 - valid loss: 1.05, valid ErrorRate: 1.14e-01
epoch: 8, lr: 8.38e-05 - train loss: 2.76 - valid loss: 1.07, valid ErrorRate: 1.30e-01
epoch: 9, lr: 8.15e-05 - train loss: 2.64 - valid loss: 1.03, valid ErrorRate: 1.22e-01
epoch: 10, lr: 7.92e-05 - train loss: 2.53 - valid loss: 9.10e-01, valid ErrorRate: 1.12e-01
epoch: 11, lr: 7.69e-05 - train loss: 2.45 - valid loss: 8.43e-01, valid ErrorRate: 1.02e-01
epoch: 12, lr: 7.46e-05 - train loss: 2.37 - valid loss: 8.29e-01, valid ErrorRate: 9.87e-02
epoch: 13, lr: 7.23e-05 - train loss: 2.30 - valid loss: 7.99e-01, valid ErrorRate: 1.01e-01
epoch: 14, lr: 7.00e-05 - train loss: 2.24 - valid loss: 7.61e-01, valid ErrorRate: 8.79e-02
epoch: 15, lr: 6.77e-05 - train loss: 2.19 - valid loss: 7.97e-01, valid ErrorRate: 9.74e-02
epoch: 16, lr: 6.54e-05 - train loss: 2.14 - valid loss: 7.20e-01, valid ErrorRate: 8.41e-02
epoch: 17, lr: 6.31e-05 - train loss: 2.09 - valid loss: 7.59e-01, valid ErrorRate: 9.68e-02
epoch: 18, lr: 6.08e-05 - train loss: 2.04 - valid loss: 7.39e-01, valid ErrorRate: 8.60e-02
epoch: 19, lr: 5.85e-05 - train loss: 2.00 - valid loss: 7.25e-01, valid ErrorRate: 8.54e-02
epoch: 20, lr: 5.62e-05 - train loss: 1.97 - valid loss: 7.48e-01, valid ErrorRate: 8.86e-02
epoch: 21, lr: 5.38e-05 - train loss: 1.92 - valid loss: 7.43e-01, valid ErrorRate: 8.73e-02
epoch: 22, lr: 5.15e-05 - train loss: 1.89 - valid loss: 6.78e-01, valid ErrorRate: 7.97e-02
epoch: 23, lr: 4.92e-05 - train loss: 1.86 - valid loss: 6.91e-01, valid ErrorRate: 8.48e-02
epoch: 24, lr: 4.69e-05 - train loss: 1.83 - valid loss: 7.45e-01, valid ErrorRate: 8.79e-02
epoch: 25, lr: 4.46e-05 - train loss: 1.80 - valid loss: 6.46e-01, valid ErrorRate: 7.97e-02
epoch: 26, lr: 4.23e-05 - train loss: 1.78 - valid loss: 6.87e-01, valid ErrorRate: 7.78e-02
epoch: 27, lr: 4.00e-05 - train loss: 1.76 - valid loss: 6.78e-01, valid ErrorRate: 8.16e-02
epoch: 28, lr: 3.77e-05 - train loss: 1.73 - valid loss: 6.68e-01, valid ErrorRate: 8.22e-02
epoch: 29, lr: 3.54e-05 - train loss: 1.71 - valid loss: 6.59e-01, valid ErrorRate: 7.21e-02
epoch: 30, lr: 3.31e-05 - train loss: 1.69 - valid loss: 6.64e-01, valid ErrorRate: 7.84e-02
epoch: 31, lr: 3.08e-05 - train loss: 1.66 - valid loss: 6.86e-01, valid ErrorRate: 7.72e-02
epoch: 32, lr: 2.85e-05 - train loss: 1.63 - valid loss: 6.59e-01, valid ErrorRate: 7.72e-02
epoch: 33, lr: 2.62e-05 - train loss: 1.52 - valid loss: 6.44e-01, valid ErrorRate: 7.46e-02
epoch: 34, lr: 2.38e-05 - train loss: 1.59 - valid loss: 6.24e-01, valid ErrorRate: 7.02e-02
epoch: 35, lr: 2.15e-05 - train loss: 1.56 - valid loss: 6.49e-01, valid ErrorRate: 7.27e-02
epoch: 36, lr: 1.92e-05 - train loss: 1.54 - valid loss: 6.11e-01, valid ErrorRate: 7.15e-02
epoch: 37, lr: 1.69e-05 - train loss: 1.52 - valid loss: 6.16e-01, valid ErrorRate: 7.02e-02
epoch: 38, lr: 1.46e-05 - train loss: 1.50 - valid loss: 6.33e-01, valid ErrorRate: 7.21e-02
epoch: 39, lr: 1.23e-05 - train loss: 1.48 - valid loss: 6.19e-01, valid ErrorRate: 6.83e-02
epoch: 40, lr: 1.00e-05 - train loss: 1.47 - valid loss: 5.82e-01, valid ErrorRate: 6.83e-02

So, your numbers are very similar to mine and would have probably reached below 7% error rate after 40 epochs.

I cleaned up the README and removed the "option 3" (train directly on the web data) as this doesn't seem to work.

Sure, I can upload the training output folder and the model to GDrive/Huggingface, but I need some write permissions?

mravanelli commented:
Great, sure I can give the permissions (I just need your email). Could you
please contact me privately for that?

On Mon, 4 Oct 2021 at 09:39, Tanel Alumäe ***@***.***> wrote:

> I actually trained for more epochs (40 instead of 30), forgot to update
> the yaml file in the repo (I did it now).
> Here is my training log:
>
> epoch: 1, lr: 1.00e-04 - train loss: 7.73 - valid loss: 3.31, valid
> ErrorRate: 2.28e-01
> epoch: 2, lr: 9.77e-05 - train loss: 5.59 - valid loss: 2.68, valid
> ErrorRate: 2.02e-01
> epoch: 3, lr: 9.54e-05 - train loss: 4.56 - valid loss: 2.13, valid
> ErrorRate: 1.87e-01
> epoch: 4, lr: 9.31e-05 - train loss: 3.87 - valid loss: 1.66, valid
> ErrorRate: 1.62e-01
> epoch: 5, lr: 9.08e-05 - train loss: 3.40 - valid loss: 1.41, valid
> ErrorRate: 1.50e-01
> epoch: 6, lr: 8.85e-05 - train loss: 3.10 - valid loss: 1.18, valid
> ErrorRate: 1.32e-01
> epoch: 7, lr: 8.62e-05 - train loss: 2.91 - valid loss: 1.05, valid
> ErrorRate: 1.14e-01
> epoch: 8, lr: 8.38e-05 - train loss: 2.76 - valid loss: 1.07, valid
> ErrorRate: 1.30e-01
> epoch: 9, lr: 8.15e-05 - train loss: 2.64 - valid loss: 1.03, valid
> ErrorRate: 1.22e-01
> epoch: 10, lr: 7.92e-05 - train loss: 2.53 - valid loss: 9.10e-01, valid
> ErrorRate: 1.12e-01
> epoch: 11, lr: 7.69e-05 - train loss: 2.45 - valid loss: 8.43e-01, valid
> ErrorRate: 1.02e-01
> epoch: 12, lr: 7.46e-05 - train loss: 2.37 - valid loss: 8.29e-01, valid
> ErrorRate: 9.87e-02
> epoch: 13, lr: 7.23e-05 - train loss: 2.30 - valid loss: 7.99e-01, valid
> ErrorRate: 1.01e-01
> epoch: 14, lr: 7.00e-05 - train loss: 2.24 - valid loss: 7.61e-01, valid
> ErrorRate: 8.79e-02
> epoch: 15, lr: 6.77e-05 - train loss: 2.19 - valid loss: 7.97e-01, valid
> ErrorRate: 9.74e-02
> epoch: 16, lr: 6.54e-05 - train loss: 2.14 - valid loss: 7.20e-01, valid
> ErrorRate: 8.41e-02
> epoch: 17, lr: 6.31e-05 - train loss: 2.09 - valid loss: 7.59e-01, valid
> ErrorRate: 9.68e-02
> epoch: 18, lr: 6.08e-05 - train loss: 2.04 - valid loss: 7.39e-01, valid
> ErrorRate: 8.60e-02
> epoch: 19, lr: 5.85e-05 - train loss: 2.00 - valid loss: 7.25e-01, valid
> ErrorRate: 8.54e-02
> epoch: 20, lr: 5.62e-05 - train loss: 1.97 - valid loss: 7.48e-01, valid
> ErrorRate: 8.86e-02
> epoch: 21, lr: 5.38e-05 - train loss: 1.92 - valid loss: 7.43e-01, valid
> ErrorRate: 8.73e-02
> epoch: 22, lr: 5.15e-05 - train loss: 1.89 - valid loss: 6.78e-01, valid
> ErrorRate: 7.97e-02
> epoch: 23, lr: 4.92e-05 - train loss: 1.86 - valid loss: 6.91e-01, valid
> ErrorRate: 8.48e-02
> epoch: 24, lr: 4.69e-05 - train loss: 1.83 - valid loss: 7.45e-01, valid
> ErrorRate: 8.79e-02
> epoch: 25, lr: 4.46e-05 - train loss: 1.80 - valid loss: 6.46e-01, valid
> ErrorRate: 7.97e-02
> epoch: 26, lr: 4.23e-05 - train loss: 1.78 - valid loss: 6.87e-01, valid
> ErrorRate: 7.78e-02
> epoch: 27, lr: 4.00e-05 - train loss: 1.76 - valid loss: 6.78e-01, valid
> ErrorRate: 8.16e-02
> epoch: 28, lr: 3.77e-05 - train loss: 1.73 - valid loss: 6.68e-01, valid
> ErrorRate: 8.22e-02
> epoch: 29, lr: 3.54e-05 - train loss: 1.71 - valid loss: 6.59e-01, valid
> ErrorRate: 7.21e-02
> epoch: 30, lr: 3.31e-05 - train loss: 1.69 - valid loss: 6.64e-01, valid
> ErrorRate: 7.84e-02
> epoch: 31, lr: 3.08e-05 - train loss: 1.66 - valid loss: 6.86e-01, valid
> ErrorRate: 7.72e-02
> epoch: 32, lr: 2.85e-05 - train loss: 1.63 - valid loss: 6.59e-01, valid
> ErrorRate: 7.72e-02
> epoch: 33, lr: 2.62e-05 - train loss: 1.52 - valid loss: 6.44e-01, valid
> ErrorRate: 7.46e-02
> epoch: 34, lr: 2.38e-05 - train loss: 1.59 - valid loss: 6.24e-01, valid
> ErrorRate: 7.02e-02
> epoch: 35, lr: 2.15e-05 - train loss: 1.56 - valid loss: 6.49e-01, valid
> ErrorRate: 7.27e-02
> epoch: 36, lr: 1.92e-05 - train loss: 1.54 - valid loss: 6.11e-01, valid
> ErrorRate: 7.15e-02
> epoch: 37, lr: 1.69e-05 - train loss: 1.52 - valid loss: 6.16e-01, valid
> ErrorRate: 7.02e-02
> epoch: 38, lr: 1.46e-05 - train loss: 1.50 - valid loss: 6.33e-01, valid
> ErrorRate: 7.21e-02
> epoch: 39, lr: 1.23e-05 - train loss: 1.48 - valid loss: 6.19e-01, valid
> ErrorRate: 6.83e-02
> epoch: 40, lr: 1.00e-05 - train loss: 1.47 - valid loss: 5.82e-01, valid
> ErrorRate: 6.83e-02
>
> So, your numbers are very similar to mine and would have probably reached
> below 7% error rate after 40 epochs.
>
> I cleaned up the README and removed the "option 3" (train directly on the
> web data) as this doesn't seem to work.
>
> Sure, I can upload the training output folder and the model to
> GDrive/Huggingface, but I need some write permissions?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/speechbrain/speechbrain/pull/950#issuecomment-933496741>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEA2ZVW2CZ2MKSMK77H47ITUFGVAZANCNFSM5CW5USUA>
> .
>

fayejf commented:
Hi @alumae Could you please have a look at 
https://huggingface.co/speechbrain/lang-id-voxlingua107-ecapa/discussions/2

I cannot reproduce 6.7% with the checkpoints on huggingface. Could you please advise if I miss anything? TIA!
